src/main/java/gyro/google/storage/BucketResource.java

```diff
@@ -0,0 +1,468 @@
+        bucket.setEncryption(getEncryption() == null ? null : getEncryption().toBucketEncryption());
+        bucket.setIamConfiguration(getIamConfiguration() == null ? null : getIamConfiguration().toBucketIamConfiguration());
+        bucket.setLifecycle(getLifecycle() == null ? null : getLifecycle().toLifecycle());
+        bucket.setLogging(getLogging() == null ? null : getLogging().toBucketLogging());
+        bucket.setRetentionPolicy(getRetentionPolicy() == null ? null : getRetentionPolicy().toBucketRententionPolicy());
+        bucket.setStorageClass(getStorageClass());
+        bucket.setVersioning(getVersioning() == null ? null : getVersioning().toBucketVersioning());
+        bucket.setWebsite(getWebsite() == null ? null : getWebsite().toBucketWebsite());
+
+        try {
+            storage.buckets().insert(getProjectId(), bucket).execute();
+        } catch (GoogleJsonResponseException e) {
+            throw new GyroException(e.getDetails().getMessage());
+        }
+    }
+
+    @Override
+    public void update(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        Storage storage = creatClient(Storage.class);
+
+        try {
+            Bucket bucket = storage.buckets().get(getName()).execute();
+            bucket.setAcl(getAcl() == null ? null : getAcl().stream().map(BucketAccessControlConfiguration::toBucketAccessControl).collect(Collectors.toList()));
+            bucket.setLabels(getLabels());
+            bucket.setLocation(getLocation());
+            bucket.setDefaultEventBasedHold(getDefaultEventBasedHold());
+            bucket.setCors(getCors().stream().map(BucketCors::toBucketCors).collect(Collectors.toList()));
+            bucket.setBilling(getBilling() == null ? null : getBilling().toBucketBilling());
+            bucket.setEncryption(getEncryption() == null ? null : getEncryption().toBucketEncryption());
+            bucket.setIamConfiguration(getIamConfiguration() == null ? null : getIamConfiguration().toBucketIamConfiguration());
+            bucket.setLifecycle(getLifecycle() == null ? null : getLifecycle().toLifecycle());
+            bucket.setLogging(getLogging() == null ? null : getLogging().toBucketLogging());
+            bucket.setRetentionPolicy(getRetentionPolicy() == null ? null : getRetentionPolicy().toBucketRententionPolicy());
+            bucket.setStorageClass(getStorageClass());
+            bucket.setVersioning(getVersioning() == null ? null : getVersioning().toBucketVersioning());
+            bucket.setWebsite(getWebsite() == null ? null : getWebsite().toBucketWebsite());
+
+            // Lock retention policy. This can not be undone and ALL assets must reach policy time to delete bucket.
+            if (changedFieldNames.contains("retention-policy")
+                    && Boolean.TRUE.equals(bucket.getRetentionPolicy().getIsLocked())) {
+                storage.buckets().lockRetentionPolicy(getName(), bucket.getMetageneration()).execute();
```

Comment:
Should throw an error if someone was trying to turn this back to ``false``.

Reply:
I'm assuming Google is doing that.